### ISSUE TITLE
HYPERLINK function invoking custom function indefinitely

### ISSUE THREAD

**Issue by lbourda** [OfficeDev/office-js#2085]
## Expected Behavior
The custom function should only be invoked once.

## Current Behavior
The custom function keeps being invoked indefinitely even though it's been seemingly completed since there's no longer #calc on the sheet, and I do get a hyperlink look-alike string.

## Steps to Reproduce, or Live Example
Apparently it's not possible to mix custom functions and buttons so you'll have to manually add the function in the sheet.

Here's the link to a simple custom function returning a string and logging to the console.
https://gist.github.com/lbourda/94e0c1dc9cc5a9b96048eb829e76b5e8

Once imported click on register, and then copy the following formula anywhere on the sheet:
=HYPERLINK(SCRIPTLAB.HYPERLINKWEIRDBEHAVIOR.FORTYTWO())

You should see the "Console" tab on the task pane being updated repeatedly.
And the state of the workbook keeps alternating between Saving and Saved.

## Context
I have a workbook containing multiple custom formulas, and in particular one inside a =HYPERLINK.
And this causes the batch process to never end, because the formula inside the =HYPERLINK keeps being invoked.

## Your Environment
* Platform [PC desktop, Mac, iOS, Office Online]: Office Online
* Host [Excel, Word, PowerPoint, etc.]: Excel
* Office version number: ______
* Operating System: Windows 10
* Browser (if using Office Online): Chrome / Edge

**Comment by grangeryy** [comment]
@lbourda , sorry for replying late. Current HYPERLINK function seems does not support to take a custom formula if you refer to [THIS](https://support.microsoft.com/en-us/office/hyperlink-function-333c7ce6-c5ae-4164-9c47-7de9b76f577f). May I know more details about your scenario? You may also provide us the sample workbook to help us understand the requirement. THX!

**Comment by lbourda** [comment]
@grangeryy, yes I do refer to [THIS](https://support.microsoft.com/en-us/office/hyperlink-function-333c7ce6-c5ae-4164-9c47-7de9b76f577f).
We use it to build a hyperlink with a formula that queries the URL based on the content of a cell reference.
I'm afraid I can't provide the workbook but it wouldn't help anyway since the query wouldn't resolve without a subscription to our services.

Note that I discovered two things while playing around.
1/ If I stop the calculation process, the cell where the hyperlink is does look like a link, with the blue color and underline, but the link does not work and the workbook keeps alternating between the saved and saving states.
2/ If the custom formula is placed in a separate cell and we reference this cell in the =HYPERLINK formula everything works fine.
So for instance if we use the previous example from Script Lab:
in A1, =SCRIPTLAB.HYPERLINKWEIRDBEHAVIOR.FORTYTWO()
in A2, =HYPERLINK(A1)
So the good news is that I've found a workaround, but it still feels like a bug to me.
